starknet/src/utils/types.cairo

```diff
@@ -150,360 +184,327 @@ impl StorageAccessFinalizationStatus of StorageAccess<FinalizationStatus> {
+    ) -> SyscallResult<()> {
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 0_u8).into()
             ),
+            offset,
             value.snapshot_timestamp
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 1_u8).into()
             ),
+            offset,
             value.start_timestamp
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 2_u8).into()
             ),
+            offset,
             value.min_end_timestamp
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 3_u8).into()
             ),
+            offset,
             value.max_end_timestamp
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 4_u8).into()
             ),
+            offset,
             value.execution_payload_hash
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 5_u8).into()
             ),
+            offset,
             value.execution_strategy
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 6_u8).into()
             ),
+            offset,
             value.author
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 7_u8).into()
             ),
+            offset,
             value.finalization_status
         );
 
-        StorageAccess::write(
+        StorageAccess::write_at_offset_internal(
             address_domain,
             storage_base_address_from_felt252(
                 storage_address_from_base_and_offset(base, 8_u8).into()
             ),
+            offset,
             value.active_voting_strategies
         )
     }
+
+    fn size_internal(value: Proposal) -> u8 {
+        9_u8
```

Comment:
Ah now I see a second implementation, is it supposed to be the number of `word`s ? (`word` being a `felt252`) ?
In that case, couldn't we pack all the `u64` to reduce storage?

Reply:
yeah we should, i assume this will be handled by the compiler/ another library soon though so not sure its worth writing our own impl